rust/src/row_set.rs

```diff
@@ -24,12 +29,107 @@ impl FFI for RowSet {
     type Origin = FromArc;
 }
 
+// Not sure about this lifetime
+impl FFI for ColumnType<'_> {
+    type Origin = FromBox;
+}
+
 #[unsafe(no_mangle)]
 pub extern "C" fn row_set_free(row_set_ptr: BridgedOwnedSharedPtr<RowSet>) {
     ArcFFI::free(row_set_ptr);
     println!("RowSet freed");
 }
 
+#[unsafe(no_mangle)]
+pub extern "C" fn row_set_get_columns_count(
+    row_set_ptr: BridgedBorrowedSharedPtr<'_, RowSet>,
+) -> usize {
+    let row_set = ArcFFI::as_ref(row_set_ptr).unwrap();
+    let pager = row_set.pager.lock().unwrap();
+    pager.column_specs().len()
+}
+
+type SetMetadata = unsafe extern "C" fn(
+    columns_ptr: ColumnsPtr,
+    value_index: usize,
+    name_ptr: *const u8,
+    name_len: usize,
+    keyspace_ptr: *const u8,
+    keyspace_len: usize,
+    table_ptr: *const u8,
+    table_len: usize,
+    type_code: usize,
+    type_info_handle: BridgedOwnedExclusivePtr<ColumnType>,
+);
```

Comment:
🔧 `ColumnType` is parametrised by a lifetime, so it's an important good practice not to omit this lifetime parameter. Please write it explicitly.

```suggestion
type SetMetadata<'typ> = unsafe extern "C" fn(
    columns_ptr: ColumnsPtr,
    value_index: usize,
    name_ptr: *const u8,
    name_len: usize,
    keyspace_ptr: *const u8,
    keyspace_len: usize,
    table_ptr: *const u8,
    table_len: usize,
    type_code: usize,
    type_info_handle: BridgedOwnedExclusivePtr<ColumnType<'typ>>,
);
```